telegram/ext/dispatcher.py

```diff
@@ -275,8 +275,10 @@ def process_update(self, update):
 
         for group in self.groups:
             try:
-                for handler in (x for x in self.handlers[group] if x.check_update(update)):
-                    handler.handle_update(update, self)
+                for handler in self.handlers[group]:
+                    check = handler.check_update(update)
+                    if check is not None and check is not False:
+                        handler.handle_update(update, self, check)
                     break
```

Comment:
I think this `break` needs one more indent. It should only break when an update is handled in a group. Not after only checking the first.